packages/main/src/plugin/kubernetes/kubernetes-port-forward-storage.ts

```diff
@@ -0,0 +1,389 @@
+   * @see UserForwardConfig
+   */
+  updateForward(config: UserForwardConfig, newConfig: UserForwardConfig): Promise<UserForwardConfig>;
+
+  /**
+   * Lists all forward configurations.
+   * @returns A list of forward configurations.
+   * @see UserForwardConfig
+   */
+  listForwards(): Promise<UserForwardConfig[]>;
+}
+
+/**
+ * Interface for file-based storage.
+ */
+export interface FileBasedStorage {
+  /**
+   * Initializes the storage.
+   * @returns Void is the operation successful.
+   */
+  initStorage(): Promise<void>;
+
+  /**
+   * Checks if the storage is initialized.
+   * @returns True if the storage is initialized, otherwise false.
+   */
+  isStorageInitialized(): boolean;
+
+  /**
+   * Gets the storage path.
+   * @returns The storage path.
+   * @throws If the storage is not initialized.
+   */
+  getStoragePath(): string;
+}
+
+/**
+ * Class representing preference folder based storage.
+ * @see FileBasedStorage
+ */
+export class PreferenceFolderBasedStorage implements FileBasedStorage {
+  private initialized: boolean = false;
+
+  /**
+   * Creates an instance of PreferenceFolderBasedStorage.
+   * @param directories - The directories instance.
+   */
+  constructor(private directories: Directories) {}
+
+  /**
+   * Gets the storage path.
+   * @returns The storage path.
+   * @throws If the storage is not initialized.
+   */
+  getStoragePath(): string {
+    if (!this.isStorageInitialized()) {
+      throw new Error('Storage has not been initialized yet.');
+    }
+    return this.getStoragePathInternal();
+  }
+
+  /**
+   * Initializes the storage.
+   * @returns Void if the operation successful.
+   * @throws If the storage is already initialized or initialization fails.
+   */
+  async initStorage(): Promise<void> {
+    if (this.initialized) {
+      throw new Error('Storage has already been initialized.');
+    }
+
+    const settingsFile = this.getStoragePathInternal();
+    const parentDirectory = path.dirname(settingsFile);
+
+    await fs.mkdir(parentDirectory, { recursive: true });
+
+    try {
+      await fs.access(settingsFile);
+    } catch (err) {
+      const error = err as NodeJS.ErrnoException;
+      if (error.code === 'ENOENT') {
+        await fs.writeFile(settingsFile, JSON.stringify([]));
+      } else {
+        throw err;
+      }
+    }
+
+    this.initialized = true;
+  }
+
+  /**
+   * Checks if the storage is initialized.
+   * @returns True if the storage is initialized, otherwise false.
+   */
+  isStorageInitialized(): boolean {
+    return this.initialized;
+  }
+
+  /**
+   * Gets the internal storage path.
+   * @returns The internal storage path.
+   */
+  protected getStoragePathInternal(): string {
+    return path.resolve(this.directories.getConfigurationDirectory(), 'port-forwards.json');
+  }
+}
+
+/**
+ * Class representing file-based configuration storage.
+ * @see ForwardConfigStorage
+ */
+export class FileBasedConfigStorage implements ForwardConfigStorage {
```

Comment:
to remove ? since there is MemoryBasedStorage ?

Reply:
It kept it so that we have most of the code when we want to switch but could be removed or add a note

Reply:
👍